include/gridtools/fn/unstructured.hpp

```diff
@@ -86,9 +86,9 @@ namespace gridtools::fn {
         template <class Tag, class Ptr, class Strides, class Domain, class Conn, class Offset>
         GT_FUNCTION constexpr auto horizontal_shift(iterator<Tag, Ptr, Strides, Domain> const &it, Conn, Offset) {
             auto const &table = host_device::at_key<Conn>(it.m_domain.m_tables);
-            auto new_index = it.m_index == -1 ? -1 : get<Offset::value>(neighbor_table::neighbors(table, it.m_index));
+            auto new_index = get<Offset::value>(neighbor_table::neighbors(table, std::max(it.m_index, 0)));
```

Comment:
Hmm. For exotic cases the table might not have an entry at 0? E.g. after an origin shift. But maybe we can document in the neighbor_table concept that the 0 index need to exist.

Reply:
No idea if there are realistic cases where this happens. And no idea if it makes sense to include this workaround. It’s currently required to have reasonable performance on Daint due to the old CUDA versions. But once we move the performance tests to Alps (we probably will, right?), the `GT_PROMISE` thing should be good enough, I guess.